mlir/include/mlir/Transforms/InliningUtils.h

```diff
@@ -141,6 +142,41 @@ class DialectInlinerInterface
     return nullptr;
   }
 
+  /// Hook to handle attribute based argument conversions that return the
+  /// conversion result or the 'argument' itself if a conversion was not needed.
+  /// The type of the returned values has to match 'targetType' and the
+  /// 'argumentAttrs' dictionary is non-null even if no attribute is present.
+  /// The hook is called after converting the callsite argument types using the
+  /// materializeCallConversion callback and right before inlining the callee
+  /// region. Any operations created using the provided 'builder' are inserted
+  /// right before the inlined callee region. Example use cases are the
+  /// insertion of copies for by value arguments or integer conversions that
+  /// require signedness information.
+  virtual Value handleArgumentConversion(OpBuilder &builder, Operation *call,
+                                         Operation *callable, Value argument,
```

Comment:
Can we really not include `CallOpInterface` and `CallableOpInterface`?

Reply:
Let's not do this. Most if not all of the stuff exposed on the interfaces is anyways already accessible via the other hook arguments. If we use the call or the callable then most likely anyways to read some attribute that is specific to the dialect and not exposed on the interface. I would thus say following the other hooks makes sense here.